_posts/2016-06-19-graphql-for-mobile.markdown

```diff
@@ -0,0 +1,99 @@
+
+This is _awesome_, because previously a [lot of this logic](https://github.com/artsy/eigen/blob/dac7c80b66b600f9a45aaae6095544fe420f0bbc/Artsy/Views/Artwork/ARArtworkActionsView.m#L310-L362) existed in a [Google Doc](https://github.com/artsy/eigen/blob/dac7c80b66b600f9a45aaae6095544fe420f0bbc/Artsy/Views/Artwork/ARArtworkActionsView.m#L108-L109) which needed to be re-implmented in 3-4 clients. On the native side we would regularly find out we were out-of-date mid-release cycle and had to rush to catch up.
+
+So, what does this mean for view models? It lessens the need for them. If you can move a lot of your derived data to the server, it is presenting the logic that should be in a view model. We've not stopped writing view models, but now discussions on them includes "should this move to Metaphysics?".
+
+#### React Native
+
+We've already [shipped one full view controller](https://twitter.com/orta/status/734880605322776576) in [React Native](https://facebook.github.io/react-native/) for our flagship app, [Eigen](https://github.com/artsy/eigen/). The advantages that came from from GraphQL were a big part of the discussion around using Reacy Native.
+
+There will be longer articles on the "why" and "how" we choose to work this way. However the TLDR: the key thing that we're excited about React Native is Relay. Using Relay, our [views](https://github.com/artsy/emission/tree/2ac6e9fc0f85ca81483bcbd6c841841104f07833/lib/components/artist) can declare a fragment of the GraphQL query that [view needs](https://github.com/artsy/emission/blob/2ac6e9fc0f85ca81483bcbd6c841841104f07833/lib/components/artist/biography.js#L60-L69).
+
+So, in our Artist View Controller, the Biography "View" ([component](https://facebook.github.io/react-native/docs/native-components-ios.html)) declares "when I am in the view heirarchy, you need to grab a `bio`, and `blurb`"
+
+``` js
+export default Relay.createContainer(Biography, {
+  fragments: {
+    artist: () => Relay.QL`
+      fragment on Artist {
+        bio
+        blurb
+      }
+    `,
+  }
+});
+```
+
+Once your views are declaring what data they need, and are acting on that data - you see less of a need to use models.
+
+---
+
+GraphQL is having a massive impact in the way that we write our apps. It means we can make much faster apps, as the network is our critical path. Faster apps means happier users, happier users means happier developers. I want to be happy. So I'm thankful that the [Web practice](https://github.com/artsy/metaphysics/graphs/contributors) gave GraphQL a try, and [welcome'd](https://github.com/artsy/metaphysics/pull/243) [us](https://github.com/artsy/metaphysics/pull/313) [to](https://github.com/artsy/metaphysics/pull/226) [the](https://github.com/artsy/metaphysics/pull/302) [party](https://github.com/artsy/metaphysics/issues/2).
```

Comment:
> It means we can make much faster apps, as the network is our critical path.

By this, do you refer to ‘mobile’ apps specifically? Because I agree with that, but I think it’s an important distinction to make.